test/mustachio/renderer_test.dart

```diff
@@ -97,191 +107,315 @@ void main() {
+    var fooTemplate = getFile('/project/foo.mustache')
+      ..writeAsStringSync('Text {{^s1}}Section{{/s1}}');
     var foo = Foo()..s1 = 'hello';
-    expect(renderFoo(foo, ast), equals('Text '));
+    expect(renderFoo(foo, fooTemplate), equals('Text '));
   });
 
   test('Renderer renders an inverted null value section node', () {
-    var parser = MustachioParser('Text {{^s1}}Section{{/s1}}');
-    var ast = parser.parse();
+    var fooTemplate = getFile('/project/foo.mustache')
+      ..writeAsStringSync('Text {{^s1}}Section{{/s1}}');
     var foo = Foo()..s1 = null;
-    expect(renderFoo(foo, ast), equals('Text Section'));
+    expect(renderFoo(foo, fooTemplate), equals('Text Section'));
   });
 
   test('Renderer resolves variable inside a value section', () {
-    var parser = MustachioParser('Text {{#foo}}{{s1}}{{/foo}}');
-    var ast = parser.parse();
+    var fooTemplate = getFile('/project/foo.mustache')
+      ..writeAsStringSync('Text {{#foo}}{{s1}}{{/foo}}');
     var bar = Bar()..foo = (Foo()..s1 = 'hello');
-    expect(renderBar(bar, ast), equals('Text hello'));
+    expect(renderBar(bar, fooTemplate), equals('Text hello'));
   });
 
   test('Renderer resolves variable from outer context inside a value section',
       () {
-    var parser = MustachioParser('Text {{#foo}}{{s2}}{{/foo}}');
-    var ast = parser.parse();
+    var fooTemplate = getFile('/project/foo.mustache')
+      ..writeAsStringSync('Text {{#foo}}{{s2}}{{/foo}}');
     var bar = Bar()
       ..foo = (Foo()..s1 = 'hello')
       ..s2 = 'goodbye';
-    expect(renderBar(bar, ast), equals('Text goodbye'));
+    expect(renderBar(bar, fooTemplate), equals('Text goodbye'));
   });
 
   test('Renderer resolves variable with key with multiple names', () {
-    var parser = MustachioParser('Text {{foo.s1}}');
-    var ast = parser.parse();
+    var fooTemplate = getFile('/project/foo.mustache')
+      ..writeAsStringSync('Text {{foo.s1}}');
     var bar = Bar()
       ..foo = (Foo()..s1 = 'hello')
       ..s2 = 'goodbye';
-    expect(renderBar(bar, ast), equals('Text hello'));
+    expect(renderBar(bar, fooTemplate), equals('Text hello'));
   });
 
   test('Renderer resolves outer variable with key with two names', () {
-    var parser = MustachioParser('Text {{#foo}}{{foo.s1}}{{/foo}}');
-    var ast = parser.parse();
+    var fooTemplate = getFile('/project/foo.mustache')
+      ..writeAsStringSync('Text {{#foo}}{{foo.s1}}{{/foo}}');
     var bar = Bar()
       ..foo = (Foo()..s1 = 'hello')
       ..s2 = 'goodbye';
-    expect(renderBar(bar, ast), equals('Text hello'));
+    expect(renderBar(bar, fooTemplate), equals('Text hello'));
   });
 
   test('Renderer resolves outer variable with key with three names', () {
-    var parser = MustachioParser('Text {{#bar}}{{bar.foo.s1}}{{/bar}}');
-    var ast = parser.parse();
+    var fooTemplate = getFile('/project/foo.mustache')
+      ..writeAsStringSync('Text {{#bar}}{{bar.foo.s1}}{{/bar}}');
```

Comment:
I'm assuming all these changes are to make these tests more consistent with the new tests that require loading partials and therefore multiple files?

Reply:
Somewhat, a little simpler.

The public API of the public render function changed from requiring an AST (List of nodes) to requiring a File, for exactly what you said: having a directory from which to load partials with relative file paths.

Mostly though, I like that the tests now don't require first running a parser, then the renderer: you just give a Template file.